chore: rephrase element templates configuration

philippfromme opened the pull request:
## What is the purpose of the change

_Briefly describe the change you are making, this helps the reviewer by providing an overview._

Rephrased documentation of configuring element templates to distinct between element templates available for all diagrams (`resouces/element-templates`) and element templates available for specific diagrams only (`.camunda/element-templates)`. This is based on feedback from our QA engineers and a follow-up to https://github.com/camunda/camunda-modeler/pull/3561.

## Are there related marketing activities

No.

## When should this change go live?

Immediately.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

philippfromme commented:
@pepopowitz I'm not sure how far back I should go and change the docs. Technically, this section first appeared in version 1.3. I've only changed `docs` and `versioned_docs/version-8.2` though.

pepopowitz commented:
> @pepopowitz I'm not sure how far back I should go and change the docs. Technically, this section first appeared in version 1.3. I've only changed `docs` and `versioned_docs/version-8.2` though.

My basic rule is: 

* if it is an improvement in language for clarity, apply to the current & next versions
* if it is correcting something that is incorrect, apply to all versions that are incorrect

In this case it seems like your changes are of the first category, and you've applied them to the correct versions.

P.S. I'm going to tag @christinaausley as a reviewer, to do a quick pass from the technical writing perspective.

pepopowitz reviewed (approved):
Looks good to me! Please await Christina's review before merging, though.

christinaausley reviewed (approved):
Technical review in latest commit 👍

nikku commented on docs/components/modeler/desktop-modeler/element-templates/configuring-templates.md:
```diff
@@ -5,20 +5,29 @@ title: Configuring templates
 
-1. **Local filesystem (Camunda Platform 7, and Camunda Platform 8):** You can fetch the JSON templates from [here](/self-managed/connectors-deployment/install-and-start.md#connector-templates). Store element templates as `.json` file in the `resources/element-templates` folder, relative to the modelers executable _or_ relative to the modeler's data directory ([see below](#example-setup)). Alterna ...
-2. **Retrieve from Cawemo (Camunda Platform 7 only):** use the [Camunda Cloud Connect Plugin for Camunda Platform 7](https://docs.camunda.org/cawemo/latest/technical-guide/integrations/modeler/) to integrate Camunda Modeler with [Cawemo](https://cawemo.com/). Camunda Modeler can then retrieve templates from catalog projects setup in Cawemo. Note that the [Cloud Connect plugin](https://docs.camund ...
+1. **Local filesystem (Camunda Platform 7 and Camunda Platform 8):** You can fetch the element templates as JSON files from [here](/self-managed/connectors-deployment/install-and-start.md#connector-templates).
```
@philippfromme Maybe we can further simplify this.

I'd love to clearly separate the different ways how the modeler finds templates from where to get C8 connector templates. My proposal is that we introduce a dedicated section `Template Discovery` with two dedicated sections (local file system, cawemo).

We can add an FAQ "fetch C8 connector templates" and move the C8 template link there.

nikku reviewed (changes requested):
Cf. comment on improved structure. Let's have a quick sync.

philippfromme commented:
I synced with @nikku and will make further adjustments to improve the documentation.

nikku commented:
@philippfromme and myself had a quick chat. To summarize we want to take another stab at this PR and clean up the general structure:

* Clearly outline that templates are discoverable from local file system
* Remove "example setup" completely
* De-prioritize Cawemo integration; it is just another way to synchronize templates (to the local file system)
* Rename "Development flow" to updating templates or something

philippfromme commented:
@nikku I simplified the documentation further, providing more helpful examples and linking the _Search paths_ documentation. I removed the _Development workflow_ section and simply mentioned reloading in the first section.

nikku commented on docs/components/modeler/desktop-modeler/element-templates/configuring-templates.md:
```diff
@@ -3,22 +3,37 @@ id: configuring-templates
+### Global templates
 
-New templates will be recognized when reconnecting to Cawemo or on Camunda Modeler reload/restart.
+For templates to be available for all diagrams store them in the `resources/element-templates` directory containing the Camunda Modeler executable. Alternatively, for element templates to be available across Camunda Modeler installations, you can store them in the `resources/element-templates` directory in the modeler's data directory.
```
```suggestion
For templates to be available for all diagrams store them in the `resources/element-templates` directory containing the Camunda Modeler executable. Alternatively, for element templates to be available across Camunda Modeler installations, you can store them in the `resources/element-templates` directory in the modeler's [data directory](../../search-paths#user-data-directory).
```

nikku commented on docs/components/modeler/desktop-modeler/element-templates/configuring-templates.md:
```diff
@@ -3,22 +3,37 @@ id: configuring-templates
 ---
 
-Element templates are defined as [JSON files](../defining-templates). There are two ways to integrate them into Camunda Modeler:
+Templates will be loaded by the Camunda Modeler at application startup. Reloading the modeler using `CtrlOrCmd+R` will also reload all templates. Templates will be treated as global or local depending on their location in your file system.
```
```suggestion
Templates will be loaded by the modeler at application startup. Reloading the app using `CtrlOrCmd+R` will also reload all templates. Templates will be treated as global or local depending on their location in your file system.
```

nikku reviewed (dismissed):
Such an improvement :tada:. Thank you!

Added two nit-pick comments that you may want to incorporate.

nikku commented on versioned_docs/version-8.2/components/modeler/desktop-modeler/element-templates/about-templates.md:
```diff
@@ -8,7 +8,7 @@ description: "Element templates are a way to extend Camunda Modeler with domain-
 :::
 
-Element templates are a way to extend the [modeler](https://camunda.org/bpmn/tool/) with domain-specific diagram elements, such as service and user tasks.
+Element templates are a way to extend the [modeler](https://camunda.com/download/) with domain-specific diagram elements, such as service and user tasks.
```
Page would likely be the modeler downloads page, however we'd need to be aware of [this](https://camunda.slack.com/archives/C034F8NA1G8/p1682497416851919).

```suggestion
Element templates are a way to extend the [modeler](https://camunda.com/download/modeler/) with domain-specific diagram elements, such as service and user tasks.
```

philippfromme replied:
Okay, will adjust.

philippfromme commented on versioned_docs/version-8.2/components/modeler/desktop-modeler/search-paths/search-paths.md:
```diff
@@ -6,16 +6,16 @@ description: "Features like element templates and plugins allow you to add your
 Features like element templates and plugins allow you to add your own resources to Desktop Modeler. For these resources to be found, they have to be in one of two directories depending on how local or global you want them to be.
 
-## App Data Directory
+## App data directory
```
Forgot about that. Will do.

philippfromme commented:
🚀

nikku commented:
@christinaausley Sorry for early merge. If there are additional changes needed to this PR, please raise your voice and we can incorporate them in a follow-up. :pray:
